### PR TITLE
feat(unlock-app) - receipts details page

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

- [x] receipt detail with supplier + purchaser information
- [x] enable edit purchaser detail for payer 
- [x] add a print button to print the receipt
- [x] fix `getReceipt` definition on `openapi.yml` 
- [x] test all flow

**HTML view (lockManager/Payer)** 

Screenshot 2023-01-26 at 15 59 30

**HTML view (non-LockManager & non-payer)**

Screenshot 2023-01-26 at 15 59 49

**Print view** 

Screenshot 2023-01-24 at 11 16 58

**Edit purchaser details** 

Screenshot 2023-01-25 at 16 11 39

**Mobile**

Screenshot 2023-01-26 at 15 44 37

Screenshot 2023-01-26 at 16 51 02

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by kalidiagne** [comment]
Update receipt card

Screenshot 2023-01-23 at 20 09 39

**Comment by julien51** [comment]
Please make sure the left column is aligned to the top as well!

**Comment by julien51** [comment]
And remove `from`.

**Comment by kalidiagne** [comment]
> Please make sure the left column is aligned to the top as well!

sorry, wdym by this? have the same distance from top for the left column?

**Comment by julien51** [comment]
Yes, have everything be aligned to the top. Now there is a big white space above `From`. (and no need to actually show the `From` word).

**Comment by kalidiagne** [comment]
> Yes, have everything be aligned to the top. Now there is a big white space above `From`. (and no need to actually show the `From` word).

Screenshot 2023-01-24 at 11 16 51

Screenshot 2023-01-24 at 11 16 58

**Comment by julien51** [comment]
For the receipt part, can we do something that looks like this:

```
Service performed                                         Amount Paid
Next Pirate King                                             123.34 CURRENCY
```

Where `CURRENCY` is replaced by the ERC20 symbol OR the base currency

**Comment by kalidiagne** [comment]
> Amount Paid



> For the receipt part, can we do something that looks like this:
> 
> ```
> Service performed                                            Amount Paid
> Next Pirate King                                             123.34 CURRENCY
> ```
> 
> Where `CURRENCY` is replaced by the ERC20 symbol OR the base currency

Screenshot 2023-01-26 at 15 59 30

**Comment by kalidiagne** [comment]
All the screen in the description section

**Comment by julien51** [comment]
Ok I think this is looking good!

**Comment by kalidiagne** [comment]
show list of transactions! cc: @julien51  

Screenshot 2023-01-27 at 17 05 04

File: unlock-app/src/components/interface/Recipts/elements/ReceiptBox.tsx [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    <div className={`flex ${inline ? 'gap-2' : 'flex-col'}`}>
+      <span className="text-base">{label}</span>
+      <div className="flex items-center">
+        <span className="block text-base font-bold break-words md:inline-block">
+          {value || '-'}
+        </span>
+      </div>
+    </div>
+  )
+}
```
Comment (searchableguy) [inline unlock-app/src/components/interface/Recipts/elements/ReceiptBox.tsx:44]: Think it's been repeated a few times in similar form. Maybe open a PR to separate it into a reusable and consistent component.

File: unlock-app/src/components/interface/Recipts/elements/ReceiptBox.tsx [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        <span className="text-base">{addressLine}</span>
+      )}
+      <span className="text-base">{country}</span>
+    </div>
+  )
+}
+
+const ReceiptBoxPlaceholder = () => {
+  return (
+    <div className="w-full max-w-lg rounded-xl bg-slate-200 animate-pulse h-[100px]"></div>
```
Comment (searchableguy) [inline unlock-app/src/components/interface/Recipts/elements/ReceiptBox.tsx:72]: Use `Placeholder` component from `@unlock-protocol/ui` please.

**Review by searchableguy — APPROVED** [review]

File: unlock-app/src/components/interface/Recipts/elements/ReceiptBox.tsx (hunk 1 above)
Comment (kalidiagne) [reply unlock-app/src/components/interface/Recipts/elements/ReceiptBox.tsx:43]: ok! ill open an issue for it!

**Pull request merged**